clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -296,6 +296,21 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
             } else {
                 this.requestTimeoutMs = config.getInt(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG);
             }
+            boolean idempotenceEnabled = config.getBoolean(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG);
+
+            if (idempotenceEnabled) {
+                this.transactionState = new TransactionState(idempotenceEnabled, time);
+                if (config.getInt(ProducerConfig.RETRIES_CONFIG) == 0) {
+                    throw new ConfigException("Need to set '" + ProducerConfig.RETRIES_CONFIG
+                            + "' to greater than zero in order to use the idempotent producer.");
```

Comment:
Might be helpful to offer a brief explanation. For example, "The producer uses sequence numbers to deduplicate messages on the broker; without retries, no duplication is possible in the first place."